src/pubsub.rs

```diff
@@ -0,0 +1,635 @@
+//! The `pubsub` module implements a threaded subscription service on client RPC request
+
+use bank::Bank;
+use bs58;
+use jsonrpc_core::futures::Future;
+use jsonrpc_core::*;
+use jsonrpc_macros::pubsub;
+use jsonrpc_pubsub::{PubSubHandler, PubSubMetadata, Session, SubscriptionId};
+use jsonrpc_ws_server::ws;
+use jsonrpc_ws_server::{RequestContext, Sender, ServerBuilder};
+use rpc::{JsonRpcRequestProcessor, RpcSignatureStatus};
+use signature::{Keypair, KeypairUtil, Signature};
+use solana_program_interface::account::Account;
+use solana_program_interface::pubkey::Pubkey;
+use std::collections::HashMap;
+use std::mem;
+use std::net::SocketAddr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{atomic, Arc, Mutex, RwLock};
+use std::thread::{sleep, Builder, JoinHandle};
+use std::time::Duration;
+
+pub enum ClientState {
+    Uninitialized,
+    Init(Sender),
+}
+
+#[derive(Serialize)]
+pub struct SubscriptionResponse {
+    pub port: u16,
+    pub path: String,
+}
+
+pub struct PubSubService {
+    _thread_hdl: JoinHandle<()>,
```

Comment:
why the `_`?  `_thread_hdl` seems used?

Reply:
I think this must be because I don't use the handle in `rpc.rs`
The compiler complains if I remove the underscore.